internal/handler/metrics.go

```diff
@@ -7,56 +7,49 @@ import (
 )
 
 func Metrics(c *gin.Context) {
-	metrics, metricsErr := metric.GetAllSystemMetrics()
-	if metricsErr != nil {
-		c.JSON(500, "Unable to get metrics")
-		return
-	}
-
+	metrics := metric.GetAllSystemMetrics()
 	c.JSON(200, metrics)
 	return
 }
 
 func MetricsCPU(c *gin.Context) {
-	metrics, metricsErr := metric.CollectCpuMetrics()
-	if metricsErr != nil {
-		c.JSON(500, "Unable to get metrics")
-		return
+	metrics, metricsErrs := metric.CollectCpuMetrics()
+	apiResponse := metric.ApiResponse{
```

Comment:
Yes that was intentional, it seemed unnecessary to add another wrapper struct around APIResponse 🤔

We could define a more suitable response struct than APIResponse, I just used it because it was there